AlphaWallet/Core/Types/RealmStore.swift

```diff
@@ -31,12 +32,18 @@ class RealmStore {
         thread.start()
     }
 
-    func performSync(_ callback: @escaping (Realm) -> Void) {
+    func performSync(_ block: @escaping (Realm) -> Void) {
```

Comment:
I don't understand the original `RealmStore.performSync(_:)` before the PR. Does it work like that:

```
func performSync(block)
    if currentThread == .mainThread
        run(block, .mainThread + mainThreadRealm)
    else
        run(block, .backgroundThread + backgroundThreadRealm)
```

Why is it optionally switching between 2 realm objects based on the caller of `performSync(:)`? Why not just one, say on .main (or background) and then always go to it?

Reply:
> Does it work like that:

yes, pretty like you described